gas/config/loongarch-parse.y

```diff
@@ -318,6 +335,7 @@ offsetT imm;
 
 primary_expression
 	: INTEGER {emit_const ($1);}
+	| IDENTIFIER {emit_const_var ($1);}
```

Comment:
Judging from this, isn't the change applicable to more insns than `li.[wd]` alone?

Reply:
You're exactly right. Let me fix this again to only works for li.w./d

Reply:
> You're exactly right. Let me fix this again to only works for li.w./d

No no no. In its current form it's way more ergonomic. I for example would be very surprised to find out that variable substitution only works for `li.[wd]`, I'd curse loudly for the anti-feature.

Reply:
I mean you'd simply write more test cases to showcase the usage with a wide variety of insns, and reword the commit message. Just don't restrict to `li` only.

Reply:
As we know,  the li.[wd] is a macro instuction used to load immediate.  li.w can load 32 bits immediate, and li.d can  load 64 bits immediate. If  a programmer need to load immediate, li.[wd] is completely competent. If other instructions also support,  for  programmer who don't know instrutions immediate bits width, It is easy to make mistakes . For eaxmple, `addi.[wd] rd, rj, var` can only load 12 bits width immediate, if `var` bits width exceeds 12, it will be not work.  So support li.[wd] is enough, that's my thought.

Reply:
I think in that case you may want to improve the error message so the user is hinted towards `li.[wd]`. Inconsistency is not good for learners, despite your good intention. It is bound to cause confusion, believe me...

Reply:
Oh, I got you, you're right. I will add  more testcases for other instructions.

Reply:
I've tested this feature of other ARCHs like x86,  their behavior is really the same as you said!!!That's cool!

Reply:
> I've tested this feature of other ARCHs like x86, their behavior is really the same as you said!!!That's cool!

Told you. ;-)